internal/proxy/proxy.go

```diff
@@ -784,7 +778,6 @@ func (c *Client) newSocketMount(ctx context.Context, conf *Config, pc *portConfi
 			version, err := c.dialer.EngineVersion(ctx, inst.Name)
 			if err != nil {
 				c.logger.Errorf("could not resolve version for %q: %v", inst.Name, err)
-				return nil, err
```

Comment:
I'm surprised we don't have test failures for removing this line. In any case, we should always be returning the error and only decide if we want to fail the proxy up above where we set up all these socket mounts.

Reply:
Agreed in principle. In this particular case, if we want to support auto reconnect after the server instance is back alive, we probably need the listening [socket](https://github.com/GoogleCloudPlatform/cloud-sql-proxy/blob/main/internal/proxy/proxy.go#L809) after this code block?